BugSplatDotNetStandard/Api/VersionsClient.cs

```diff
@@ -0,0 +1,153 @@
+        /// Get a list of applications and versions with crashes and/or symbols for a BugSplat database
+        /// </summary>
+        /// <param name="database">BugSplat database for which version information will be retrieved</param>
+        public async Task<HttpResponseMessage> GetVersions(string database)
+        {
+            ThrowIfArgumentIsNullOrEmpty(database, "database");
+
+            ThrowIfNotAuthenticated(bugsplatApiClient);
+
+            var response = await this.bugsplatApiClient.GetAsync($"/api/versions?database={database}");
+
+            ThrowIfHttpRequestFailed(response);
+
+            return response;
+        }
+
+        /// <summary>
+        /// Upload a symbol file to BugSplat and authenticate if not already authenticated
+        /// </summary>
+        /// <param name="database">BugSplat database to upload symbols too</param>
+        /// <param name="application">BugSplat application value to associate with the symbol file</param>
+        /// <param name="version">BugSplat version value to associate with the symbol file</param>
+        /// <param name="symbolFileInfo">The symbol file that will be uploaded to BugSplat</param>
+        public async Task<HttpResponseMessage> UploadSymbolFile(
+            string database,
+            string application,
+            string version,
+            FileInfo symbolFileInfo
+        )
+        {
+            ThrowIfArgumentIsNullOrEmpty(database, "database");
+            ThrowIfArgumentIsNullOrEmpty(application, "application");
+            ThrowIfArgumentIsNullOrEmpty(version, "version");
+            ThrowIfArgumentIsNull(symbolFileInfo, "symbolFileInfo");
+
+            ThrowIfNotAuthenticated(bugsplatApiClient);
+
+            var random = Path.GetRandomFileName();
+            var zipFileName = $"bugsplat-{symbolFileInfo.Name}-{random}.zip";
+            var zipFileFullName = Path.Combine(Path.GetTempPath(), zipFileName);
+            using (var zipFile = ZipFile.Open(zipFileFullName, ZipArchiveMode.Create))
+            {
+                zipFile.CreateEntryFromFile(symbolFileInfo.FullName, symbolFileInfo.Name);
+            }
+
+            using (var zipFileStream = new FileStream(zipFileFullName, FileMode.Open))
+            using (
+                var presignedUrlResponse = await this.GetSymbolUploadUrl(
+                    database,
+                    application,
+                    version,
+                    symbolFileInfo.Length,
+                    symbolFileInfo.Name
+                )
+            )
+            {
+                ThrowIfHttpRequestFailed(presignedUrlResponse);
+
+                var presignedUrl = await this.GetPresignedUrlFromResponse(presignedUrlResponse);
+                var uploadFileResponse = await s3Client.UploadFileStreamToPresignedURL(presignedUrl, zipFileStream);
+                
+                ThrowIfHttpRequestFailed(uploadFileResponse);
+
+                return uploadFileResponse;
+            }
```

Comment:
Delete the zip file you dork!

Reply:
Can we add a test for this?!

Reply:
[48d4630](https://github.com/BugSplat-Git/bugsplat-net-standard/pull/52/commits/48d46301baebef718c9200b4fb007169586e6ed0)